Enables to add a kinc dll project as a visual studio subproject

mundusnine opened the pull request:
In the root project kincfile add : 
![image](https://user-images.githubusercontent.com/29337013/145600653-808441c5-849a-4401-9c87-29ff65bb25d3.png)

To add as a dynamic library subproject.

RobDangerous commented:
I reverted it because everything stopped working. Also formatting was totally off (use tslint to check the basics). Also the PR was just too big, I have no idea why it has to do so many things.

mundusnine commented:
> I reverted it because everything stopped working. Also formatting was totally off (use tslint to check the basics). Also the PR was just too big, I have no idea why it has to do so many things.

I just  installed tslint and added change on autosave. It didn't work because the RaytraceApi stuff was removed, I made the change but when I dev I always have a version that I test and then I copy paste to a fork, but I didn't see that I left in that [part...](https://github.com/foundry2D/kincmake/blob/172a959c9f8624564d884ef4d2fb71153b9920e2/src/Project.ts#L120)

 Regarding the PR being big:
1. well it didn't resolve paths of the .lib when it was only `Debug\\nameofthedll.lib`. I don't know if that ever worked as we don't actually use .lib really afaik. So my fix was to use the absolute path.
2. Since subprojects inherit their options from the main project we need to bypass it to enable having dll's as instead of putting the vcxproj type as DynamicLibrary, it would put the vcxproj type as an Application. We could remove the for loop if we added a field to Project which would remember the option specified for libraries.
